.buildkite/rebuild.hs

```diff
@@ -141,17 +142,37 @@ buildStep dryRun bk nightly = do
     pkgs <- listLocalPackages
     let cabalFlags = concatMap (flag "release") pkgs
 
-    titled "Build LTS Snapshot"
-        (build Standard ["--only-snapshot"]) .&&.
-      titled "Build dependencies"
-        (build Standard ["--only-dependencies"]) .&&.
-      titled "Build"
-        (build Fast (["--test", "--no-run-tests"] ++ cabalFlags)) .&&.
-      titled "Test"
-        (timeout 60 (test Fast cabalFlags)) .&&.
-      titled "Checking golden test files"
-        (checkUnclean dryRun "lib/core/test/data")
+    let shouldRunIntegration = case qaLevel nightly bk of
+            QuickTest -> False
+            FullTest -> True
+            NightlyTest -> True
+
+    let justWhen pred x = if pred then Just x else Nothing
+
+    foldl1 (.&&.) $ catMaybes
+        [ pure $ titled "Build LTS Snapshot"
+            (build Standard ["--only-snapshot"])
+        , pure $ titled "Build dependencies"
+            (build Standard ["--only-dependencies"])
+        , pure $ titled "Build"
+            (build Fast (["--test", "--no-run-tests"] ++ cabalFlags))
+        , pure
+            $ titled "Tests (except integration)"
+            $ timeout 60
+            $ test Fast
+            $ cabalFlags <> skip "integration"
+        , justWhen shouldRunIntegration $ titled "Integration tests on latest era"
+            $ timeout 60
+            $ integration Fast cabalFlags
+        , justWhen shouldRunIntegration $ titled "Integration tests on past era (Mary)"
+            $ timeout 60
+            $ ((export "LOCAL_CLUSTER_ERA" "mary") >>)
```

Comment:
Should ensure the logs (`TESTS_LOGSDIR`) don't overwrite the Alonzo logs.

Reply:
It does seem like the artefacts in buildkite are currently only for a single run.

Reply:
Should be fixed now.